### ISSUE TITLE
Documentation on re-writing does not behave as expected

### ISSUE THREAD

**Issue by Andersen98** [JuliaSymbolics/SymbolicUtils.jl#420]
I walked through the documentation [here](https://github.com/JuliaSymbolics/SymbolicUtils.jl/blob/master/page/rewrite.md) 

`r1(sin(2*(w-z)))` does not match and `r3 = @rule ~x * +(~~ys) => sum(map(y-> ~x * y, ~~ys)); r3(2 * (w+w+α+β))` does not match. I switched over to Metatheory.jl and it seems to work perfectly though.

**Comment by 0x0f0f0f** [comment]
The doc over here should probably be updated

**Comment by anstadnik** [comment]
Hi! Can it be resolved?
The only solution to figure out why @rule doesn't work as expected now is to read the code and commit messages.
I have a similar problem, and there is no place for me to start from.

**Comment by anstadnik** [comment]
[This](https://github.com/JuliaSymbolics/SymbolicUtils.jl/issues/426#issuecomment-1017101487) might be related

**Comment by karlwessel** [comment]
This seems to have been fixed at some point, at least for me on 3.7 it gives the documented output:
```julia
julia> using SymbolicUtils
julia> @syms w z α::Real β::Real
(w, z, α, β)

julia> r1 = @rule sin(2(~x)) => 2sin(~x)*cos(~x)
sin(2 * ~x) => (2 * sin(~x)) * cos(~x)
julia> r1(sin(2*(w-z)))
2cos(w - z)*sin(w - z)

julia> r3 = @rule ~x * +(~~ys) => sum(map(y-> ~x * y, ~~ys));
julia> r3(2 * (w+w+α+β))
4w + 2α + 2β
```
